usr/src/uts/common/io/drm/drm_sunmod.c

```diff
@@ -236,20 +265,30 @@ drm_gem_map_access(devmap_cookie_t dhp, void *pvt, offset_t offset, size_t len,
 	struct gem_map_list *seg;
 
 	obj = (struct drm_gem_object *)pvt;
-	if (obj == NULL) {
-		dev = NULL;	/* avoid "used uninitialized" warning */
-		goto next;
-	}
 
-	dev = obj->dev;
-	if (dev->driver->gem_fault != NULL)
-		dev->driver->gem_fault(obj);
+	/*
+	 * Only call the driver fault handler after gtt_map_kaddr
```

Comment:
BTW, there previously was a more convoluted mechanism to skip these calls to devmap_load, where __devmap_gem() would not even install these devmap callbacks if the back-end resources have not been setup.
I'd rather have the callbacks always setup to make sure we have a chance to do this ref management on the GEM object, so __devmap_gem() now always installs the callbacks, and we just skip gem_fault when appropriate.

Let me know if you think that design changes needs more commentary somewhere.